src/heartbeat/heartbeat.js

```diff
@@ -46,7 +41,8 @@ module.exports = () => {
     collectorConfig: changed,
   };
 
-  return httpUtils.doPost(urlToPost, collectorToken, proxy, requestbody, cutoff)
+  return httpUtils.doPost(urlToPost, cr.collectorToken, cr.proxy, requestbody,
+    cutoff)
   .then((res) => listener(null, res.body))
   .catch((err) => listener(err, null));
```

Comment:
Since we have a separate onError function now, it might make this more clear to expose that and call it directly here.